src/basic/process-util.c

```diff
@@ -1680,6 +1683,16 @@ int safe_fork_full(
                 }
         }
 
+        if (FLAGS_SET(flags, FORK_SETSID) && setsid() < 0) {
+                log_full_errno(prio, errno, "Failed to create a new session: %m");
+                _exit(EXIT_FAILURE);
+        }
+
+        if (FLAGS_SET(flags, FORK_TIOCSCTTY) && ioctl(STDERR_FILENO, TIOCSCTTY) < 0) {
```

Comment:
Why STDERR btw? I figure the new helper should check whether STDOUT and STDERR are the same fd connected to terminal

Reply:
I don't see the harm in using `STDERR_FILENO` here. I don't think this is a massive footgun that requires extra checks to prevent misuse either.

If we have a helper to verify stdin and stdout point to the same tty I'm happy to use it but I'd like to not write it from scratch.

Reply:
btw, i think we should use stdin here, see https://github.com/systemd/systemd/pull/35811#pullrequestreview-2527810810 for my rationale